lib/ramble/ramble/repository.py

```diff
@@ -590,120 +682,125 @@ def repo_for_app(self, spec):
                 tty.debug('Found repo...')
                 return repo
 
-        # If the application isn't in any repo, return the one with
+        # If the object isn't in any repo, return the one with
         # highest precedence.  This is for commands like `ramble edit`
-        # that can operate on applications that don't exist yet.
+        # that can operate on objects that don't exist yet.
         return self.first_repo()
 
     @autospec
     def get(self, spec):
-        """Returns the application associated with the supplied spec."""
-        return self.repo_for_app(spec).get(spec)
+        """Returns the object associated with the supplied spec."""
+        return self.repo_for_obj(spec).get(spec)
 
-    def get_app_class(self, app_name):
-        """Find a class for the spec's application and return the class object."""  # noqa: E501
-        return self.repo_for_app(app_name).get_app_class(app_name)
+    def get_obj_class(self, obj_name):
+        """Find a class for the spec's object and return the class object."""  # noqa: E501
+        return self.repo_for_obj(obj_name).get_obj_class(obj_name)
 
     @autospec
     def dump_provenance(self, spec, path):
         """Dump provenance information for a spec to a particular path.
 
            This dumps the package file and any associated patch files.
-           Raises UnknownApplicationError if not found.
+           Raises UnknownObjectError if not found.
         """
-        return self.repo_for_app(spec).dump_provenance(spec, path)
+        return self.repo_for_obj(spec).dump_provenance(spec, path)
 
-    def dirname_for_application_name(self, app_name):
-        return self.repo_for_app(app_name).dirname_for_application_name(
-            app_name)
+    def dirname_for_object_name(self, obj_name):
```

Comment:
This is a bit of a vague comment, and for that I apologies.. but ...

We're already in a class designed around the access/storage of objects (the non-OOP type), and we use the word `object` _a lot_. For example in this method we call it `dirname_for_object_name` and pass it and `object_name`.. maybe it should just be `dirname` or `get_dirname` ? 

That makes me think a lot of the object stuff is actually already strongly implied by being in this class (it gets passed an object_type at init, too..)? 
 
When we make one we're pretty explicit
```
app_path = RepoPath(*repo_dirs, object_type=ObjectTypes.applications)
```

and when we call things:

```
return self.repo_for_obj(obj_name).dirname_for_object_name(
```
Maybe this could be quite a lot simpler?

```
return self.repo_for_obj(obj_name).dirname(
```

Reply:
I don't disagree at all, but I would consider that as part of a re-design with the spack folks more than just in ramble here.

I'm still "trying" to stay somewhat aligned with how they do stuff until we have some sort of shared infrastructure.